File: plugin/src/main/groovy/com/github/gfx/ribbonizer/YellowRibbonBuilder.java
package com.github.gfx.ribbonizer;

import com.android.build.gradle.api.ApplicationVariant;
import com.github.gfx.ribbonizer.filter.ColorRibbonFilter;

import java.awt.*;
import java.awt.image.BufferedImage;
import java.util.function.Consumer;

public class YellowRibbonBuilder implements FilterBuilder {

    @Override
    public Consumer<BufferedImage> apply(ApplicationVariant variant) {
        return new ColorRibbonFilter(variant.getBuildType().getName(), new Color(0xff, 0x76, 0, 0xaa));
    }
}

File: ribbonizer-plugin/src/main/kotlin/com/shogo82148/ribbonizer/YellowRibbonBuilder.kt
package com.shogo82148.ribbonizer

import com.android.build.api.variant.ApplicationVariantBuilder
import com.shogo82148.ribbonizer.filter.ColorRibbonFilter
import com.shogo82148.ribbonizer.resource.Resource
import java.awt.Color
import java.io.File
import java.util.function.Consumer

class YellowRibbonBuilder : FilterBuilder {
    companion object {
        val COLOR: Color = Color(0xff, 0x76, 0, 0x99)
    }

    override fun apply(variant: ApplicationVariantBuilder, iconFile: File): Consumer<Resource> {
        return ColorRibbonFilter(variant.buildType ?: "", COLOR)
    }
}